src/astro/databases/base.py

```diff
@@ -250,3 +276,61 @@ def export_table_to_file(
 
         df = self.export_table_to_pandas_dataframe(source_table)
         target_file.create_from_dataframe(df)
+
+    # ---------------------------------------------------------
+    # Schema Management
+    # ---------------------------------------------------------
+
+    def create_schema_if_needed(self, schema) -> None:
+        """
+        This function checks if the expected schema exists in the database. If the schema does not exist,
+        it will attempt to create it.
+
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        # We check if the schema exists first because snowflake will fail on a create schema query even if it
+        # doesn't actually create a schema.
+        if schema and not self.schema_exists(schema):
+            statement = self._create_schema_statement.format(schema)
+            self.run_sql(statement)
+
+    def schema_exists(self, schema) -> bool:
+        """
+        Checks if a schema exists in the database
+
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        raise NotImplementedError
+
+    # ---------------------------------------------------------
+    # Context & Template Rendering methods (Transformations)
+    # ---------------------------------------------------------
+
+    def get_sqlalchemy_template_table_identifier_and_parameter(
+        self, table: Table, jinja_table_identifier: str
+    ) -> Tuple[str, str]:
+        """
+        During the conversion from a Jinja-templated SQL query to a SQLAlchemy query, there is the need to
+        convert a Jinja table identifier to a safe SQLAlchemy-compatible table identifier.
+
+        For example, the query:
+            sql_statement = "SELECT * FROM {{input_table}};"
+            parameters = {"input_table": Table(name="user_defined_table", metadata=Metadata(schema="some_schema"))}
+
+        Can become (depending on the database):
+            "SELECT * FROM some_schema.user_defined_table;"
+            parameters = {"input_table": "user_defined_table"}
+
+        Since the table value is templated, there is a safety concern (e.g. SQL injection).
+        We recommend looking into the documentation of the database and seeing what are the best practices.
+        For example, Snowflake:
+        https://docs.snowflake.com/en/sql-reference/identifier-literal.html
+
+        :param table: The table object we want to generate a safe table identifier for
+        :param jinja_table_identifier: The name used within the Jinja template to represent this table
+        :return:
```

Comment:
Missed description on return

Reply:
```suggestion
        :return: value to replace the table identifier in the query and the value that should be used to replace it
```

Found it one of the databases